### ISSUE TITLE
[Bug]: When Running same code in headless is true , output differs as compared to headless is false

### ISSUE THREAD

**Issue by SSV0726** [puppeteer/puppeteer#10510]
### Minimal, reproducible example

```TypeScript
browser = await chromium.puppeteer.launch({
            args: chromium.args,
            slowMo: 20 ,
            defaultViewport: chromium.defaultViewport,
            executablePath: await chromium.executablePath,
            headless: true,
            ignoreHTTPSErrors: true
        });
```

### Error string

Different behavior

### Bug behavior

- [X] Flaky
- [ ] PDF

### Background
"chrome-aws-lambda": "^10.1.0",      "puppeteer": "^19.11.1",     "puppeteer-core": "^10.4.0",    
I was logging in to a website , which then redirects to another website . 

When Headless is true :
Its not getting redirected , feels like its stucking in the loop ,

When Headless is false :
 It works correctly and redirects to the desired website . 

### Expectation

When Headless is true :
It should behave in same manner as , when headless is false . 

### Reality

When Headless is true :
Its not getting redirected , feels like its stucking in the loop .

### Puppeteer configuration

```TypeScript
browser = await chromium.puppeteer.launch({
            args: chromium.args,
            slowMo: 20 ,
            defaultViewport: chromium.defaultViewport,
            executablePath: await chromium.executablePath,
            headless: true,
            ignoreHTTPSErrors: true
        });
```

### Puppeteer version

10.4.0

### Node version

v16.3.0

### Package manager

npm

### Package manager version

7.15.1

### Operating system

macOS

**Comment by Lightning00Blade** [comment]
Please use the latest version of Puppeteer. Also don't mix Puppeteer and Puppeteer-core version as they are published together so Puppeteer `20.7.4` requires Puppeteer-core `20.7.4`. For differences in headless to headful you can read more about it here -> https://developer.chrome.com/articles/new-headless/.
Also provide a code that this behavior is observed and can be tested from our side - [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example).

**Comment by SSV0726** [comment]
Not a bug , i verified that the website which i was trying to call , didn't allow headless browser to operate on it .
